rollup.config.js

```diff
@@ -34,6 +34,7 @@ export const appConfiguration = {
     replace({
       CLIENT_ID: JSON.stringify(process.env.CLIENT_ID),
       API_KEY: JSON.stringify(process.env.API_KEY),
+      DEVELOPMENT: process.env?.DEVELOPMENT === 'true'
```

Comment:
TL;DR: remove the `?`
```suggestion
      DEVELOPMENT: process.env.DEVELOPMENT === 'true'
```

The optional _chaining_ operator is intended for uses cases where the _earlier_ parts might be undefined or null, e.g.,
 
```javascript
      // `process` might be undefined:
      DEVELOPMENT: process?.env.DEVELOPMENT === 'true'
```

If the end part (`.DEVELOPMENT`) is `null` or `undefined`, regular old JavaScript checks still apply.

---

The Vercel build error is due to this line. I'm not sure why you can't see the logs, but I've copy-pasted them here:

```

00:38:12.291 | Cloning github.com/eddieantonio/predictive-text-studio (Branch: feature/dev-logging, Commit: ed58edc)
-- | --
00:38:12.726 | Cloning completed in 435ms
00:38:12.733 | Analyzing source code...
00:38:13.267 | Installing build runtime...
00:38:16.364 | Build runtime installed: 3096.753ms
00:38:18.883 | Looking up build cache...
00:38:19.354 | Build cache found. Downloading...
00:38:22.367 | Build cache downloaded [41.36 MB]: 3011.786ms
00:38:23.207 | Installing dependencies...
00:38:23.506 | yarn install v1.22.10
00:38:23.609 | [1/4] Resolving packages...
00:38:23.919 | success Already up-to-date.
00:38:23.930 | Done in 0.43s.
00:38:24.417 | yarn run v1.22.10
00:38:24.477 | $ rollup -c
00:38:24.682 | [!] SyntaxError: Unexpected token '.'
00:38:24.683 | /vercel/workpath0/rollup.config.js:50
00:38:24.683 | DEVELOPMENT: process.env?.DEVELOPMENT === 'true'
00:38:24.684 | ^
00:38:24.684 | SyntaxError: Unexpected token '.'
00:38:24.684 | at wrapSafe (internal/modules/cjs/loader.js:915:16)
00:38:24.684 | at Module._compile (internal/modules/cjs/loader.js:963:27)
00:38:24.684 | at Object.require.extensions.<computed> [as .js] (/vercel/workpath0/node_modules/rollup/dist/shared/loadConfigFile.js:511:20)
00:38:24.684 | at Module.load (internal/modules/cjs/loader.js:863:32)
00:38:24.684 | at Function.Module._load (internal/modules/cjs/loader.js:708:14)
00:38:24.684 | at Module.require (internal/modules/cjs/loader.js:887:19)
00:38:24.684 | at require (internal/modules/cjs/helpers.js:74:18)
00:38:24.684 | at loadConfigFromBundledFile (/vercel/workpath0/node_modules/rollup/dist/shared/loadConfigFile.js:519:42)
00:38:24.684 | at getDefaultFromTranspiledConfigFile (/vercel/workpath0/node_modules/rollup/dist/shared/loadConfigFile.js:503:12)
00:38:24.684 | at loadConfigFile (/vercel/workpath0/node_modules/rollup/dist/shared/loadConfigFile.js:481:15)
00:38:24.699 | error Command failed with exit code 1.
00:38:24.699 | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
00:38:24.709 | Error: Command "yarn run build" exited with 1
00:38:29.285 | Done with "package.json"
```

The reason this fails is because whatever version of Node that Vercel is using does not support the ES2020 [optional chaining operator](https://github.com/tc39/proposal-optional-chaining). See this table:

https://node.green/#ES2020-features-optional-chaining-operator-----

[Vercel uses the latest Node LTS version](https://vercel.com/docs/runtimes#official-runtimes/node-js/node-js-version) which _just_ changed to 14.0/.

---

EDIT: after digging, it turns out I _explicity_ set the Node version to 12.x:

![Screen Shot 2021-02-23 at 11 22 06 AM](https://user-images.githubusercontent.com/2294397/108889324-7a8efe80-75c9-11eb-98fd-5ef8a49493b3.png)

I changed it to 14.x LTS 😅

Reply:
Thanks for the insightful response and investigation! Looks like I got ahead of myself with the optional chaining 😅. I'm not sure why vercel is acting up for me, but I'll chalk it up as a one-off. If it happens again I'll go into a bit more depth as to what's acting up.

Reply:
Lol, Vercel was just using an older version of Node until I changed it, haha 😓